src/windows.rs

```diff
@@ -0,0 +1,480 @@
+            0,
+            out_buffer.as_mut_ptr() as *mut c_void,
+            out_buffer.capacity() as u32,
+            &mut bytes_returned,
+            std::ptr::null_mut(),
+        )
+        .ok()
+        .map(|_| {
+            out_buffer.set_len(bytes_returned as usize);
+            IoctlSupportedBrightnessLevels(out_buffer)
+        })
+        .map_err(|e| e.code())
+    }
+}
+
+fn ioctl_query_display_brightness(device: &Brightness) -> Result<u32, SysError> {
+    unsafe {
+        let mut bytes_returned = 0;
+        let mut display_brightness = DISPLAY_BRIGHTNESS::default();
+        DeviceIoControl(
+            device.file_handle.0,
+            IOCTL_VIDEO_QUERY_DISPLAY_BRIGHTNESS,
+            std::ptr::null_mut(),
+            0,
+            (&mut display_brightness as *mut DISPLAY_BRIGHTNESS) as *mut c_void,
+            std::mem::size_of::<DISPLAY_BRIGHTNESS>() as u32,
+            &mut bytes_returned,
+            std::ptr::null_mut(),
+        )
+        .ok()
+        .map(|_| {
+            if display_brightness.ucDisplayPolicy == DISPLAYPOLICY_AC as u8 {
+                // This is a value between 0 and 100.
+                display_brightness.ucACBrightness as u32
+            } else if display_brightness.ucDisplayPolicy == DISPLAYPOLICY_DC as u8 {
+                // This is a value between 0 and 100.
+                display_brightness.ucDCBrightness as u32
+            } else {
+                panic!("Unexpected ucDisplayPolicy");
+            }
+        })
+        .map_err(|e| SysError::GettingBrightnessFailed {
+            device_name: device.device_name.clone(),
+            function_name: "IOCTL_VIDEO_QUERY_DISPLAY_BRIGHTNESS",
+            last_error: e.code(),
+        })
+    }
+}
+
+fn ioctl_set_display_brightness(device: &Brightness, value: u8) -> Result<(), SysError> {
+    // Seems to currently be missing from metadata
+    const DISPLAYPOLICY_BOTH: u8 = 3;
+    unsafe {
+        // TODO: Consider whether this is actually the desired behaviour?
+        let mut display_brightness = DISPLAY_BRIGHTNESS {
+            ucACBrightness: value,
+            ucDCBrightness: value,
+            ucDisplayPolicy: DISPLAYPOLICY_BOTH,
+        };
+        let mut bytes_returned = 0;
+        DeviceIoControl(
+            device.file_handle.0,
+            IOCTL_VIDEO_SET_DISPLAY_BRIGHTNESS,
+            (&mut display_brightness as *mut DISPLAY_BRIGHTNESS) as *mut c_void,
+            std::mem::size_of::<DISPLAY_BRIGHTNESS>() as u32,
+            std::ptr::null_mut(),
+            0,
+            &mut bytes_returned,
+            std::ptr::null_mut(),
+        )
+        .ok()
+        .map(|_| {
+            // There is a bug where if the IOCTL_VIDEO_QUERY_DISPLAY_BRIGHTNESS is
+            // called immediately after then it won't show the newly updated values
+            // Doing a very tiny sleep seems to mitigate this
+            std::thread::sleep(std::time::Duration::from_nanos(1));
```

Comment:
I really dislike using `sleep` to work around bugs, but I don't have other suggestions. I can reproduce the issue you described.

`IOCTL_VIDEO_QUERY_DISPLAY_BRIGHTNESS` supposedly does not work since Windows Vista so I guess we're lucky it returns any meaningful value at all.

Reply:
Yeah, it is indeed ugly. But my concern is that seeing strange results (such as when running the example program), is far more unexpected for the user. 
The sleep is so small it is basically just a yield, like any other system call, which is a reasonably expected behaviour IMHO.

Reply:
Let me know if you would rather me leave it in, or remove it, it doesn't make any difference for my personal usage of the library. It just makes the example look broken without it

Reply:
Let's keep it. I can't see anything in our code or the documentation that explains this bug, so it is more likely caused by some bug/oddity in Windows itself.